datadotworld/__init__.py

```diff
@@ -35,7 +35,7 @@
 )
 from datadotworld.datadotworld import DataDotWorld, UriParam  # noqa: F401
 
-__version__ = '1.8.5'
+__version__ = '2.0.0a'
```

Comment:
updating this to 2.0.0a, since this is backwards-incompatible (removes support for Python < 3.9)

"a" because this really is the minimal set of changes I could find to successfully build on python 3.12 - there's a bunch more modernization we COULD do here, and we may want to do at least a little of that before committing to 2.0.0 proper.